src/main/java/com/listywave/list/application/service/CommentService.java

```diff
@@ -0,0 +1,75 @@
+package com.listywave.list.application.service;
+
+import static java.util.function.Function.identity;
+import static java.util.stream.Collectors.toMap;
+
+import com.listywave.auth.application.domain.JwtManager;
+import com.listywave.list.application.domain.Comment;
+import com.listywave.list.application.domain.Lists;
+import com.listywave.list.application.domain.Reply;
+import com.listywave.list.application.dto.response.CommentCreateResponse;
+import com.listywave.list.application.dto.response.CommentFindResponse;
+import com.listywave.list.repository.CommentRepository;
+import com.listywave.list.repository.ListRepository;
+import com.listywave.list.repository.ReplyRepository;
+import com.listywave.user.application.domain.User;
+import com.listywave.user.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class CommentService {
+
+    private final JwtManager jwtManager;
+    private final ListRepository listRepository;
+    private final UserRepository userRepository;
+    private final ReplyRepository replyRepository;
+    private final CommentRepository commentRepository;
+
+    public CommentCreateResponse create(Long listId, String accessToken, String content) {
+        Long userId = jwtManager.read(accessToken);
+        User user = userRepository.getById(userId);
+        Lists list = listRepository.getById(listId);
+
+        Comment comment = Comment.create(list, user, content);
+        Comment saved = commentRepository.save(comment);
+
+        return CommentCreateResponse.of(saved, user);
+    }
+
+    public CommentFindResponse getComments(Long listId, int size, Long cursorId) {
+        Lists list = listRepository.getById(listId);
+        Long totalCount = commentRepository.countByList(list);
+
+        List<Comment> comments = commentRepository.getComments(listId, size, cursorId);
+        Map<Comment, List<Reply>> result = comments.stream()
+                .collect(toMap(
+                        identity(),
+                        replyRepository::getAllByComment
+                ));
```

Comment:
네 확인해보니 해당 toMap의 결과는 순서가 뒤죽박죽 될 우려가 크네요
해당 부분을 순서를 보장받고 싶으면 이렇게 작성하시면 좋을 거 같아요👍🏼

https://ttl-blog.tistory.com/1232  
해당 글 참고해보세요!

```suggestion
        List<Comment> comments = commentRepository.getComments(listId, size, cursorId);
        Map<Comment, List<Reply>> result = comments.stream()
                .collect(toMap(
                        identity(),
                        replyRepository::getAllByComment,
                        (existing, replacement) -> existing, // 중복 키 발생시 새로 유입된 키로 덮으겠다.
                        LinkedHashMap::new  // LinkedhashMap 구현체를 사용하여 맵에 삽입된 순서 보장
                ));
```

Reply:
앗 (existing, replacement) -> replacement로 바꿔야합니다!

Reply:
오 첨부해주신 링크의 블로그 주인이 저랑 같이 우테코에서 프로젝트 했던 팀원인데 신기하네요 ㅋㅋㅋ
감사합니다 👍

Reply:
원래 세상은 좁아요 ㅎ